packages/vertexai/src/requests/request.ts

```diff
@@ -84,6 +84,7 @@ export async function getHeaders(url: RequestUrl): Promise<Headers> {
   headers.append('Content-Type', 'application/json');
   headers.append('x-goog-api-client', getClientHeaders());
   headers.append('x-goog-api-key', url.apiSettings.apiKey);
+  headers.append('X-Firebase-AppId', url.apiSettings.appId); // Will be converted to 'X-Firebase-Appid' before it's sent in the browser.
```

Comment:
Does JS have a global dataCollection flag? The appID should only be added to the header if dataCollection is enabled.

Header names are case insensitive https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive

Reply:
Yes, it's `FirebaseApp.automaticDataCollectionEnabled`. Does this apply to the `x-goog-api-client` header too or is that ok to log without this being true?

Reply:
Fixed in https://github.com/firebase/firebase-js-sdk/pull/8809/commits/767cd7717b8487852a0aaf8dac1daca13e64c501.

Discussed in sync that `x-goog-api-client` does not need to be gated behind `automaticDataCollectionEnabled`.